src/sentry/notifications/platform/target.py

```diff
@@ -92,3 +119,42 @@ def integration_installation(self) -> IntegrationInstallationT:
             model=self.integration,
             organization_id=self.organization_integration.organization_id,
         )
+
+
+@dataclass
+class SerializedTargetType:
```

Comment:
Attempt at a wrapper class whose only job is to handle serializing and de-serializing a `NotificationTarget`